### PR TITLE
Update auth to reflect introduction of alternate back-end keys

### PR DESCRIPTION
**Pull request by keirog**
(See changes in [config-vars](https://github.com/Financial-Times/next-config-vars/pull/188) and [n-express](https://github.com/Financial-Times/n-express/pull/342))

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -3,17 +3,29 @@
...
 
+	if (process.env.FT_NEXT_BACKEND_KEY) {
+		backendKeys.push(process.env.FT_NEXT_BACKEND_KEY);
+	}
+	if (process.env.FT_NEXT_BACKEND_KEY_OLD) {
+		backendKeys.push(process.env.FT_NEXT_BACKEND_KEY_OLD);
+	}
+	if (process.env.FT_NEXT_BACKEND_KEY_OLDEST) {
+		backendKeys.push(process.env.FT_NEXT_BACKEND_KEY_OLDEST);
+	}
```
Comment (phamann) [inline lib/auth.js]:
```
Object.keys(process.env)
    .filter(key => /FT_NEXT_BACKEND_KEY/.test(key))
    .forEach(key =>  backendKeys.push(process.env[key]));
```

File: lib/auth.js [hunk 2]
```diff
@@ -3,17 +3,29 @@
...
+
 	const validUser = event['FT-User-Uuid'] === event.uuid;
-	const validBackend = event['FT-Next-Backend-Key'] === process.env['FT_NEXT_BACKEND_KEY'];
+	const validBackend = backendKeys.indexOf(event['FT-Next-Backend-Key']) > -1;
```
Comment (phamann) [inline lib/auth.js]:
```js
backendKeys.some(key => key === event['FT-Next-Backend-Key']);
```

**Comment by phamann** [comment]
👍

File: lib/auth.js (hunk 1 above)
Comment (keirog) [inline lib/auth.js]: Much nicer

File: lib/auth.js (hunk 2 above)
Comment (keirog) [inline lib/auth.js]: Aha, thanks! I was distracted by my disappointment at not being able to user `.includes`

**Pull request merged**